src/test/java/com/rakovets/course/java/core/practice/string/StringUtilTest.java

```diff
@@ -0,0 +1,289 @@
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class StringUtilTest {
+    static Stream<Arguments> concatProviderArguments() {
+        return Stream.of(
+                Arguments.of("HelloJava", "Hello", "Java"),
+                Arguments.of("Java", "", "Java"),
+                Arguments.of("Hello", "Hello", "")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("concatProviderArguments")
+    void Glue(String expectedString, String str1, String str2) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.glue(str1, str2);
+
+        // THAT
+        Assertions.assertEquals(expectedString, actualString);
+    }
+
+    static Stream<Arguments> indexSymbol() {
+        return Stream.of(
+                Arguments.of(2, "HelloJava", "l"),
+                Arguments.of(2, "Java", "v"),
+                Arguments.of(1, "Symbol", "y")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("indexSymbol")
+    void IndexSymbol(int expectedSymbol, String str1, char symbol) {
+        // GIVEN
+
+        // WHEN
+        int actualString = StringUtil.indexSymbol(str1, symbol);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> identicalStrings() {
+        return Stream.of(
+                Arguments.of("true", "Hello", "Hello"),
+                Arguments.of("false", "Hello", "hello"),
+                Arguments.of("true", " ", " ")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("identicalStrings")
+    void identicalStrings(boolean expectedSymbol, String str1, String str2) {
+        // GIVEN
+
+        // WHEN
+        boolean actualString = StringUtil.identicalStrings(str1, str2);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> removeSpacesAroundCastUpperCase() {
+        return Stream.of(
+                Arguments.of("HELLO", "Hello "),
+                Arguments.of("HELLO", " HeLlo"),
+                Arguments.of("", " ")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("removeSpacesAroundCastUpperCase")
+    void removeSpacesAroundCastUpperCase(String expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.removeSpacesAroundCastUpperCase(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> extractionSubstring() {
+        return Stream.of(
+                Arguments.of("ll", "Hello  ", 2, 4),
+                Arguments.of("l", " HeLlo", 4, 5),
+                Arguments.of("lati", "Relationship", 2, 6)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("extractionSubstring")
+    void extractionSubstring(String expectedSymbol, String str1, int nSymbol, int mSymbol) {
```

Comment:
"Violation: 'nameMethodTest' was not used for test-method naming
Solution: Rename method
"